src/hooks/screen/useCreatePostHook.js

```diff
@@ -12,27 +12,22 @@ import AnonUserInfoRepo from '../../service/repo/anonUserInfoRepo';
 const useCreatePostHook = (isAnonymous) => {
   const {params = {}} = useRoute();
   const {topic} = params;
+  const [selectedTopic, setSelectedTopic] = React.useState(topic);
   const [anonUserInfo, setAnonUserInfo] = React.useState(null);
+  const [headerTitle, setHeaderTitle] = React.useState(
+    selectedTopic ? `Create Post in #${selectedTopic}` : 'Create Post'
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useState` hook for `selectedTopic` is initialized with the value of `topic`, but then `selectedTopic` is used immediately in the next `useState` hook for `headerTitle`. This could lead to an incorrect initial state for `headerTitle` because state updates in React may be asynchronous, and `selectedTopic` might not have been updated when it's used. To fix this, you can directly use `topic` instead of `selectedTopic` when setting the initial state for `headerTitle`.

```diff
- const [headerTitle, setHeaderTitle] = React.useState(
-   selectedTopic ? `Create Post in #${selectedTopic}` : 'Create Post'
- );
+ const [headerTitle, setHeaderTitle] = React.useState(
+   topic ? `Create Post in #${selectedTopic}` : 'Create Post'
+ );
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->